### PR TITLE
[wicket] Add live update status for artifacts

### PR DESCRIPTION
**Pull request by andrewjstone**
* Parse UpdateLogAll to update state
* Start some small styling changes based on discussion Friday. More to come.

### CODE REVIEW

File: wicket/src/state/update.rs [hunk 1]
```diff
@@ -103,6 +127,162 @@ impl RackUpdateState {
...
+                    }
+                    UpdateNormalEventKind::InstallinatorEvent(
+                        _completion_event_kind,
+                    ) => {
+
+                        // TODO: Do we want to update state
+                        // with intermediate events?
+                        //
+                        // This would be useful for reporting progress
+                        // percentages, but that's about it.
```
Comment (sunshowers) [inline wicket/src/state/update.rs:175]: Can we represent "downloaded" and "installed" as separate events here? That would fit nicely here then, I think

File: wicket/src/state/update.rs [hunk 2]
```diff
@@ -103,6 +127,162 @@ impl RackUpdateState {
...
+            }
+            SettingHostPowerState { .. }
+            | InstallinatorFormatProgress { .. }
+            | SettingInstallinatorOptions
+            | SettingHostStartupOptions => {
+                // Should we bother doing something here?
+                None
+            }
+        };
+        update_artifact_state(items, known, UpdateState::Updating);
```
Comment (sunshowers) [inline wicket/src/state/update.rs:239]: Per discussion, filed https://github.com/oxidecomputer/omicron/issues/2578 to show a progress percentage here.

**Review by sunshowers — APPROVED** [review]

File: wicket/src/state/update.rs (hunk 1 above)
Comment (andrewjstone) [reply wicket/src/state/update.rs:175]: Maybe. One problem is that this doesn't really work for the Sp and RoT which write while they are downloading. It's all streaming. I'll merge this through and we can discuss in a follow up. There's also what @jgallagher suggested which is in the status column to indicate for sleds which thing was being updated. I didn't do that just due to the current structure of the code where `UpdateState` is an enum and all states are currently valid for all entities. But that could change and may line up with this here.

File: wicket/src/state/update.rs (hunk 2 above)
Comment (andrewjstone) [reply wicket/src/state/update.rs:239]: Thank you!

**Pull request merged**